src/Fmt_ast.ml

```diff
@@ -2792,18 +2794,18 @@ and fmt_constructor_declaration c ctx ~first ~last:_ cstr_decl =
     ( match c.conf.type_decl with
     | `Sparse -> "@;<1000 0>"
     | `Compact -> "@ " )
-  $ Cmts.fmt_before c.cmts pcd_loc
-  $ Cmts.fmt_before c.cmts loc
+  $ Cmts.fmt_before c pcd_loc $ Cmts.fmt_before c loc
   $ fmt_or_k first (if_newline "| ") (fmt "| ")
   $ hvbox 0
       ( hovbox 2
           ( hvbox 2
-              ( Cmts.fmt c.cmts loc
+              ( Cmts.fmt c loc
                   (wrap_if (is_symbol_id txt) "( " " )" (str txt))
               $ fmt_constructor_arguments_result c ctx pcd_args pcd_res )
           $ fmt_attributes c ~pre:(fmt "@;") ~key:"@" atrs
           $ fmt_docstring_padded c doc )
-      $ Cmts.fmt_after c.cmts pcd_loc )
+      $ Cmts.fmt_after c ~pro:(fmt_or c.conf.wrap_comments "@ " " ") pcd_loc
```

Comment:
given the size of the PR, should this be moved to a different PR ?

Reply:
I agree, done.